docs/source/csdl-spec.mdx

```diff
@@ -0,0 +1,319 @@
+
+```graphql
+type Rocket 
+  @key(fields: "{ id }", graph: "rockets")
+  @key(fields: "{ id }", graph: "astronauts") 
+{
+	id: String! 
+  name: String!
+	astronaut: Astronaut!
+  captain: Astronaut @provides(fields: "{ tripId }")
+}
+```
+
+### `@owner`
+
+`directive @owner(graph: String!) on OBJECT`
+
+Just like any other GraphQL service, federated services can define any types they would like or need. When a service defines a type that hasn't been defined before (that is, it's not extending a type), then that service is referred to as the `owner` of that type. Type ownership is important, because services have the ability to extend types defined by other services in the graph, and those extenders can require additional fields from the owner to help them resolve their extensions. In our example, the `Astronaut` type is defined by the astronauts service, and the `Rocket`  type by the rockets service. We can represent that in CSDL like so:
+
+```graphql
+type Astronaut 
+	@key(fields: "{ id }", graph: "astronauts")
+	@key(fields: "{ id }", graph: "rockets")
+	@owner(graph: "astronauts")
+{
+  id: String!
+  name: String!
+  rocket: Rocket!
+	tripId: String!
+}
+
+type Rocket 
+	@key(fields: "{ id }", graph: "rockets")
+	@key(fields: "{ id }", graph: "astronauts")
+	@owner(graph: "rockets")
+{
+  id: String!
+  name: String!
+	astronaut: Astronaut!
+  captain: Astronaut @provides(fields: "{ tripId }")
+}
+```
+
+**Not all types** get an `@owner` defined in CSDL:
+- Value types are shared amongst all services and don't receive an owner.
+- Types which are defined and used only in one service don't receive an owner.
+- Root types (`Query` and `Mutation`) must not have owners.
+
+
+### `@resolve`
+
+`directive @resolve(graph: String!) on FIELD_DEFINITION`
+
+Once we know about owning services for types, we have to represent what service is responsible for resolving fields that are extended by other services. We do this with the `resolve` directive. For the `rocket` field on our `Astronaut` type and the `astronaut` field on our `Rocket` type, that would look like this:
+
+```graphql
+type Astronaut 
+	@key(fields: "{ id }", graph: "astronauts")
+	@key(fields: "{ id }", graph: "rockets")
+	@owner(graph: "astronauts")
+{
+  id: String!
+  name: String!
+	tripId: String!
+  rocket: Rocket! @resolve(graph: "rockets")
+}
+
+type Rocket 
+	@key(fields: "{ id }", graph: "rockets")
+	@key(fields: "{ id }", graph: "astronauts")
+	@owner(graph: "rockets")
+{
+  id: String!
+  name: String!
+  astronaut: Astronaut! @resolve(graph: "astronauts")
+	captain: Astronaut @provides(fields: "{ tripId }")
+}
+```
+
+In addition to these fields, all fields on the root `Query` or `Mutation` types also need a `@resolve` directive, since those types aren't owned by any service
+
+```graphql
+type Query {
+  rockets: [Rocket]! @resolve(graph: "rockets")
+  astronauts: [Astronaut]! @resolve(graph: "astronauts")
+}
+```
+
+### `@csdl` and `@graph`
```

Comment:
I want to change `@composedGraph(version:)` to `@csdl(version:)` bc it feels like it more clearly indicates that it's the version of CSDL this schema requires, rather than the version of this composed graph.